lib/model.js

```diff
@@ -233,6 +233,21 @@ Model.prototype._tableWasCreated = function() {
 };
 
 /*
+ * Return a promise which resolves when/if the Model is ready to use
+ */
+Model.prototype._hasTable = function() {
+  var self = this;
+  if (this._getModel()._tableCreated) {
+    return new Promise.resolve();
+  }
+  return new Promise(function(resolve) {
+    self.on('created', function() {
```

Comment:
This should probably be `self.once('created', resolve)` so that the listener gets destroyed afterwards.